common/interfaces.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed
+ * under the Apache License Version 2.0.
+ *
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2021 Datadog, Inc.
+ */
+
+export interface DatadogProps {
+  readonly pythonLayerVersion?: number;
+  readonly nodeLayerVersion?: number;
+  readonly extensionLayerVersion?: number;
+  readonly addLayers?: boolean;
+  readonly forwarderArn?: string;
+  readonly flushMetricsToLogs?: boolean;
+  readonly site?: string;
+  readonly apiKey?: string;
+  readonly apiKeySecretArn?: string;
+  readonly apiKmsKey?: string;
+  readonly enableDatadogTracing?: boolean;
+  readonly injectLogContext?: boolean;
+  readonly logLevel?: string;
+  readonly enableDatadogLogs?: boolean;
+  readonly captureLambdaPayload?: boolean;
+}
+
+/*
+ * Makes fields shared with DefaultDatadogProps (in constants file) required.
+ *
+ * Cannot use "extends DatadogProps" or "extends Omit<DatadogProps, "addLayers" | "enableDatadogLogs"...>"
+ * Due to JSII typescript restrictions.
+ *
+ * TODO: Research better way to create this second interface
+ */
+export interface DatadogStrictProps {
+  readonly addLayers: boolean;
+  readonly enableDatadogLogs: boolean;
+  readonly captureLambdaPayload: boolean;
+  readonly injectLogContext: boolean;
+  readonly enableDatadogTracing: boolean;
+  readonly pythonLayerVersion?: number;
+  readonly nodeLayerVersion?: number;
+  readonly extensionLayerVersion?: number;
+  readonly forwarderArn?: string;
+  readonly flushMetricsToLogs?: boolean;
+  readonly site?: string;
+  readonly apiKey?: string;
+  readonly apiKeySecretArn?: string;
+  readonly apiKmsKey?: string;
+  readonly logLevel?: string;
+}
+
+export interface ILambdaFunction {
+  runtime: any;
+  node: any;
+  addEnvironment(ey: string, value: string, options?: Record<string, unknown>): unknown;
+}
```

Comment:
This is the interface imported in place of an AWS CDK specific import (`import * as lambda from "@aws-cdk/aws-lambda";`)

Reply:
is it possible to get real types instead of any?

Reply:
`runtime` and `node` are classes, who's fields are primarily subclasses [heres the api reference for the Function class from aws](https://docs.aws.amazon.com/cdk/api/v2//docs/aws-cdk-lib.aws_lambda.Function.html)

I think we should keep it as is?

Reply:
Alternative:
```
export interface Runtime {
  readonly name: string
}

export interface Node {
  readonly defaultChild: any;
}

export interface ILambdaFunction {
  runtime: Runtime;
  node: Node;
  addEnvironment(ey: string, value: string, options?: Record<string, unknown>): void;
}
```

Reply:
I was having issues with JSII, so I wasn't able implement a real type for Node.default child.

I have implemented this alternative